examples/chart/teleport-cluster/values.yaml

```diff
@@ -363,6 +365,37 @@ gcp:
   # You can override this to a blank value if the worker node running Teleport already has a service account which grants access.
   credentialSecretName: teleport-gcp-credentials
 
+#####################################################
+# Azure-specific settings (only used in "azure" mode)
+#####################################################
+azure:
+  # The hostname of the Postgres database cluster hosted in Azure. It must not contain the
+  # suffix ".postgres.database.azure.com".
+  databaseHost: ""
+  # The Postgres user Teleport must use to connect to the backend and audit
+  # databases.
+  databaseUser: ""
+  # The Postgres database to use for backend storage.
+  backendDatabase: "teleport_backend"
+  # The Postgres database to use for audit log storage.
+  # This MUST NOT be the same database as used for 'backendDatabase'.
+  auditLogDatabase: "teleport_audit"
+  # Whether to mirror audit log entries to stdout in JSON format (useful for external log collectors)
+  auditLogMirrorOnStdout: false
+  # The name of the Azure Blob Storage account to use for recorded session storage. This account must already exist.
+  sessionRecordingStorageAccount: ""
+  # Azure client ID is used by the Kubernetes Service Account to know which
+  # Application it should impersonate. This can be unset only if the clientID is
+  # passed through other means (e.g. environment variable)
```

Comment:
Refer that this is not required if the pod only has one identity associated.

Reply:
Won't the pod always have access to the kubelet identity first?

Reply:
I don't think so. Similarly to GKE and EKS, AKS should allow one to restrict pods from inheriting the node's permissions by blocking the metadata if the number of hops is bigger than one.

https://learn.microsoft.com/en-us/azure/aks/operator-best-practices-cluster-security?tabs=azure-cli#restrict-access-to-instance-metadata-api

Reply:
On GKE and AWS I think you end up using ambiant credentials if you don't set anything. On GKE they used to offer a lot of privileges, now I think you have almost no access by default.

According to your docs, the metadata API is not restricted by default (and it's not on EKS as well).

Reply:
Yes, it's not restricted by default, but it's highly recommended to restrict the traffic. 
That recommendation is part of the best practices in every cloud provider

Reply:
Even with a restricted egress to the IDMS, in this test I just did the pod gets a failure to obtain credentials; maybe because the admission webhook used by azure still adds an AZURE_CLIENT_ID envvar that's just the empty string. 😩